src/Infrastructure/Identity/UserService.cs

```diff
@@ -170,4 +173,14 @@ public async Task ToggleUserStatusAsync(ToggleUserStatusRequest request, Cancell
         user.IsActive = request.ActivateUser;
         var identityResult = await _userManager.UpdateAsync(user);
     }
+
+    public async Task<bool> IsRootTenantAsync(string userId, CancellationToken cancellationToken)
+    {
+        var user = await _userManager.Users.Where(u => u.Id == userId).FirstOrDefaultAsync(cancellationToken);
```

Comment:
Why are you loading the user from the db again? That has just been done right before calling this method...